src/iter.rs

```diff
@@ -85,3 +86,158 @@ fn hex_chars_to_byte(hi: u8, lo: u8) -> Result<u8, HexToBytesError> {
     let ret = (hih << 4) + loh;
     Ok(ret as u8)
 }
+
+/// Iterator over bytes which encodes the bytes and yields hex characters.
+pub struct BytesToHexIter<I: ExactSizeIterator + FusedIterator + Iterator<Item = u8>> {
+    /// The iterator whose next byte will be encoded to yield hex characters.
+    iter: I,
+    /// The low character of the pair (high, low) of hex characters encoded per byte.
+    low: Option<char>,
+}
+
+impl<I> BytesToHexIter<I>
+where
+    I: ExactSizeIterator + FusedIterator + Iterator<Item = u8>,
+{
+    /// Constructs a new `BytesToHexIter` from a byte iterator.
+    pub fn new(iter: I) -> BytesToHexIter<I> { Self { iter, low: None } }
+}
+
+impl<I> Iterator for BytesToHexIter<I>
+where
+    I: ExactSizeIterator + FusedIterator + Iterator<Item = u8>,
+{
+    type Item = char;
+
+    fn next(&mut self) -> Option<char> {
+        match self.low {
+            Some(c) => {
+                self.low = None;
+                Some(c)
+            }
+            None => self.iter.next().map(|b| {
+                let (high, low) = byte_to_hex_chars(b);
+                self.low = Some(low);
+                high
+            }),
+        }
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) { (self.len(), Some(self.len())) }
```

Comment:
`self.len()` normally calls into `size_hint`, not the other way around. It's not wrong but might be a bit better to keep the standard.